python/pyplugin_installer/installer.py

```diff
@@ -408,7 +408,7 @@ def uninstallPlugin(self, key, quiet=False):
             return
         if not quiet:
             warning = self.tr("Are you sure you want to uninstall the following plugin?") + "\n(" + plugin["name"] + ")"
-            if plugin["status"] == "orphan" and not plugin["error"]:
+            if plugin["status"] == "orphan" and plugin["status_exp"] == "orphan" and not plugin["error"]:
```

Comment:
Is it guaranteed that the status_exp key will always exist? Or would a .get be more appropriate here?

Reply:
From my investigation, yes.
This `status_exp` will be set either in https://github.com/qgis/QGIS/blob/master/python/pyplugin_installer/installer_data.py#L447 (for the plugin from the plugin repository) or https://github.com/qgis/QGIS/blob/master/python/pyplugin_installer/installer_data.py#L686 (for the installed plugin).

Other parts of the code also do not use `.get` like https://github.com/qgis/QGIS/blob/master/python/pyplugin_installer/installer_data.py#L810 (and the following lines) or https://github.com/qgis/QGIS/blob/master/python/pyplugin_installer/installer.py#L222

Reply:
Ok, thanks for the confirmation!